rmw_connext_cpp/src/rmw_client.cpp

```diff
@@ -0,0 +1,338 @@
+// Copyright 2014-2017 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+
+#include "rmw/allocators.h"
+#include "rmw/error_handling.h"
+#include "rmw/impl/cpp/macros.hpp"
+#include "rmw/rmw.h"
+
+#include "rmw_connext_shared_cpp/types.hpp"
+#include "rmw_connext_shared_cpp/qos.hpp"
+
+#include "identifier.hpp"
+#include "process_topic_and_service_names.hpp"
+#include "type_support_commons.hpp"
```

Comment:
apologies for being late to the party, but are the any objections if we change "commons" to "common"? I thought it was a typo the first time I saw it. I can make a followup PRs